### PR TITLE
unit test for LoadBalanceController

### PR DESCRIPTION
**Pull request by sunbufu**

### CODE REVIEW

File: dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    public void createLoadbalance() throws IOException {
+        BalancingDTO balancingDTO = new BalancingDTO();
+        ResponseEntity<String> response;
+
+        // service and application are all blank
+        response = restTemplate.postForEntity(url("/api/{env}/rules/balancing"), balancingDTO, String.class, env);
```
Comment (kezhenxu94) [inline dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java]: Why not specify the return type as `Boolean.class`? Then there is no need to use `ObjectMapper`, you can use `responseEntity.getBody()`

File: dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        assertFalse("should return a fail response, when service and application are all blank", (Boolean) objectMapper.readValue(response.getBody(), Map.class).get("success"));
+        // dubbo version is 2.6
+        balancingDTO.setApplication("test application");
+        balancingDTO.setService("test service");
+        when(providerService.findVersionInApplication("test application")).thenReturn("2.6");
+        response = restTemplate.postForEntity(url("/api/{env}/rules/balancing"), balancingDTO, String.class, env);
+        assertFalse("should return a fail response, when dubbo version is 2.6", (Boolean) objectMapper.readValue(response.getBody(), Map.class).get("success"));
```
Comment (kezhenxu94) [inline dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java]: Why not specify the return type as Boolean.class? Then there is no need to use ObjectMapper, you can use responseEntity.getBody()

File: dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        // dubbo version is 2.7
+        when(providerService.findVersionInApplication("test application")).thenReturn("2.7");
+        response = restTemplate.postForEntity(url("/api/{env}/rules/balancing"), balancingDTO, String.class, env);
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+        assertTrue(Boolean.valueOf(response.getBody()));
```
Comment (kezhenxu94) [inline dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java]: Why not specify the return type as Boolean.class? Then there is no need to use ObjectMapper, you can use responseEntity.getBody()

File: dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        // unknown id
+        uri = restTemplate.getUriTemplateHandler().expand(url("/api/{env}/rules/balancing/{id}"), env, id);
+        response = restTemplate.exchange(uri, HttpMethod.PUT, new HttpEntity<>(balancingDTO), String.class);
+        assertFalse("should return a fail response, when id is null", (Boolean) objectMapper.readValue(response.getBody(), Map.class).get("success"));
+        // valid id
+        BalancingDTO balancing = mock(BalancingDTO.class);
+        when(overrideService.findBalance(id)).thenReturn(balancing);
+        uri = restTemplate.getUriTemplateHandler().expand(url("/api/{env}/rules/balancing/{id}"), env, id);
+        response = restTemplate.exchange(uri, HttpMethod.PUT, new HttpEntity<>(balancingDTO), String.class);
+        assertTrue(Boolean.valueOf(response.getBody()));
```
Comment (kezhenxu94) [inline dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java]: Why not specify the return type as Boolean.class? Then there is no need to use ObjectMapper, you can use responseEntity.getBody()

File: dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        // application is valid
+        response = restTemplate.getForEntity(url("/api/{env}/rules/balancing?service={service}&application={application}"), String.class, env, null, application);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        verify(overrideService).findBalance(application);
+        // findBalance return a notnull
+        BalancingDTO balancingDTO = new BalancingDTO();
+        when(overrideService.findBalance(anyString())).thenReturn(balancingDTO);
+        response = restTemplate.getForEntity(url("/api/{env}/rules/balancing?service={service}&application={application}"), String.class, env, null, application);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(1, objectMapper.readValue(response.getBody(), List.class).size());
```
Comment (kezhenxu94) [inline dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java]: Why not specify the return type as List.class? Then there is no need to use ObjectMapper, you can use responseEntity.getBody()

File: dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    @Test
+    public void deleteLoadBalance() {
+        String id = "1";
+        URI uri;
+        ResponseEntity<String> response;
+
+        uri = restTemplate.getUriTemplateHandler().expand(url("/api/{env}/rules/balancing/{id}"), env, id);
+        response = restTemplate.exchange(uri, HttpMethod.DELETE, new HttpEntity<>(null), String.class);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertTrue(Boolean.valueOf(response.getBody()));
```
Comment (kezhenxu94) [inline dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java]: Same as above

**Comment by nzomkxia** [comment]
please check travis CI

**Pull request merged**